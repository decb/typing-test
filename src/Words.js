import React from "react";

import { addWord, nWords } from "./textGenerator";
import UserInput from "./UserInput";

const wordsSize = 8;

function Words(props) {
  const [lower, setLower] = React.useState(0);
  const [words, setWords] = React.useState(() => nWords(props.data, wordsSize));

  const setCount = n => {
    setLower(n);
    if (lower + wordsSize === words.length) {
      setWords(addWord(props.data, words));
    }
  }

  const visible = words.slice(lower, lower + wordsSize).join(" ");

  return (
    <div>
      <code>{visible}</code>
      <br />
      <UserInput setCount={setCount} />
    </div>
  );
}

export default Words;
