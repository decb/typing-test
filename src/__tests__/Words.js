import React from "react";
import renderer from "react-test-renderer";

import Words from "../Words";

const progress = "a";
const words = ["ab"];
const range = [0, 1];

describe("<Words />", () => {
  test("Renders correctly", () => {
    const tree = renderer
      .create(<Words progress={progress} words={words} range={range} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
