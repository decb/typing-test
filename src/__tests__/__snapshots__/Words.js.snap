// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Words /> Renders correctly 1`] = `
<div
  style={
    Object {
      "overflow": "hidden",
      "whiteSpace": "nowrap",
    }
  }
>
  <code>
    <span
      style={
        Object {
          "backgroundColor": "#ffe58f",
          "fontWeight": "bold",
          "padding": "3px",
        }
      }
    >
      ab
    </span>
     
    
  </code>
</div>
`;
